lib/kubeclient.rb

```diff
@@ -52,10 +52,15 @@ def initialize(uri,
       @headers = {}
       @ssl_options = ssl_options
 
-      @basic_auth_user = auth_options[:user]
-      @basic_auth_password = auth_options[:password]
-
-      bearer_token(auth_options[:bearer_token]) if auth_options[:bearer_token]
+      if auth_options[:user]
+        @basic_auth_user = auth_options[:user]
+        @basic_auth_password = auth_options[:password]
+      elsif auth_options[:bearer_token]
+        bearer_token(auth_options[:bearer_token])
+      elsif auth_options[:bearer_token_file] &&
+            File.file(auth_options[:bearer_token_file])
+        bearer_token(File.read(auth_options[:bearer_token_file]))
```

Comment:
So wrong token will be handled as before - returning 403s, but if the file doesn't exist or is unreadable do you think this should return an error in the constructor? I think it should so let me update & push that change too. If you think differently let me know & I will update accordingly.